### PR TITLE
Line item module

### PR DESCRIPTION
**Pull request by amitshethe**

### CODE REVIEW

File: app/controllers/line_items_controller.rb [hunk 1]
```diff
@@ -1,52 +1,59 @@
...
       @line_item.quantity += 1
     else
       @line_item = LineItem.new
       @line_item.cart = current_cart
       @line_item.product = chosen_product
-      @line_item.quantity = 1      
+      @line_item.quantity = 1
+    end
+    begin
+      @line_item.save!
```
Comment (nyadav-deqode) [inline app/controllers/line_items_controller.rb]: What is the need to use save!.
Can't we use
if line_item.save
  #redirect
else
  #show error
end

File: app/controllers/line_items_controller.rb [hunk 2]
```diff
@@ -1,52 +1,59 @@
...
-  end  
-    
+  end
+
   def add_quantity
     @line_item.quantity ||= 0
     @line_item.quantity += 1
-    @line_item.save!
-    redirect_to cart_path(@current_cart)
+    if @line_item.update_attribute(:quantity, @line_item.quantity)
```
Comment (nyadav-deqode) [inline app/controllers/line_items_controller.rb]: use update_attributes.
Check difference between update_attribute, update_column and update_attributes

File: app/controllers/line_items_controller.rb [hunk 3]
```diff
@@ -1,52 +1,59 @@
...
+    end
   end
-      
+
   def reduce_quantity
-    if @line_item.quantity > 1
-      @line_item.quantity -= 1
+    @line_item.quantity -= 1 if @line_item.quantity > 1
+    if @line_item.update_attribute(:quantity, @line_item.quantity)
+      redirect_to cart_path(@current_cart)
```
Comment (nyadav-deqode) [inline app/controllers/line_items_controller.rb]: Same as above

**Review by nyadav-deqode — CHANGES_REQUESTED** [review]

File: app/controllers/line_items_controller.rb (hunk 1 above)
Comment (amitshethe) [reply app/controllers/line_items_controller.rb]: yes we can use save but just to practice exception handling I have used save!.
In next update I will change it to save

Comment (amitshethe) [reply app/controllers/line_items_controller.rb]: updated with save instead of save!

File: app/controllers/line_items_controller.rb (hunk 2 above)
Comment (amitshethe) [reply app/controllers/line_items_controller.rb]: used update instead of update_attribute

File: app/controllers/line_items_controller.rb (hunk 3 above)
Comment (amitshethe) [reply app/controllers/line_items_controller.rb]: used update instead of update_attribute

File: app/controllers/line_items_controller.rb [hunk 4]
```diff
@@ -1,52 +1,58 @@
...
+      @line_item = current_cart.line_items.find_by(product_id: chosen_product)
       @line_item.quantity += 1
     else
       @line_item = LineItem.new
       @line_item.cart = current_cart
       @line_item.product = chosen_product
-      @line_item.quantity = 1      
+      @line_item.quantity = 1
+    end
+    if @line_item.save
```
Comment (nyadav-deqode) [inline app/controllers/line_items_controller.rb]: The code before line_item.save can be moved into a separate method and can be called here

**Review by nyadav-deqode — APPROVED** [review]

**Pull request merged**